third_party/asio-grpc/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+#[[
+   Copyright 2025 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+]]
+
+set(ASIO_GRPC_USE_BOOST_CONTAINER YES)
+
+add_subdirectory(asio-grpc)
```

Comment:
`add_subdirectory(asio-grpc SYSTEM)` would have spared you from all those GCC pragmas to ignore warnings. Additionally, I am happy to fix those https://github.com/Tradias/asio-grpc/issues/123